packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -38,6 +38,10 @@ const Tolerance _kFlingTolerance = const Tolerance(
   distance: 0.01,
 );
 
+/// Don't start the animation if the target value is within this tolerance of
+/// the current value.
+const Tolerance _kAnimateToTolerance = const Tolerance();
```

Comment:
You want to get this from the ScrollPosition somehow. Maybe AnimationController should take a Tolerance that it then passes on to all its simulations, etc, and uses in animateTo; then you could just pass the ScrollPosition's tolerance through (I forget where exactly we store that, it might be on the ScrollPhysics).